subsys/nfc/lib/platform.c

```diff
@@ -43,21 +38,21 @@ ISR_DIRECT_DECLARE(nfc_isr_wrapper)
 	return 1;
 }
 
-static void clock_handler(struct onoff_manager *mgr, int res)
-{
-	/* Activate NFCT only when HFXO is running */
-	nrfx_nfct_state_force(NRFX_NFCT_STATE_ACTIVATED);
-}
-
 nrfx_err_t nfc_platform_setup(nfc_lib_cb_resolve_t nfc_lib_cb_resolve)
 {
 	int err;
 
-	hf_mgr = z_nrf_clock_control_get_onoff(CLOCK_CONTROL_NRF_SUBSYS_HF);
-	__ASSERT_NO_MSG(hf_mgr);
+	err = nfc_platform_internal_hfclk_init();
+	__ASSERT_NO_MSG(err == 0);
 
+#ifdef CONFIG_NFC_ZERO_LATENCY_IRQ
+	IRQ_DIRECT_CONNECT(NFCT_IRQn, CONFIG_NFCT_IRQ_PRIORITY,
+			   nfc_isr_wrapper, IRQ_ZERO_LATENCY);
+#else
 	IRQ_DIRECT_CONNECT(NFCT_IRQn, CONFIG_NFCT_IRQ_PRIORITY,
 			   nfc_isr_wrapper, 0);
+#endif /* CONFIG_NFC_ZERO_LATENCY_IRQ */
```

Comment:
```suggestion
IRQ_DIRECT_CONNECT(NFCT_IRQn, CONFIG_NFCT_IRQ_PRIORITY,
			   nfc_isr_wrapper, NFCT_IRQ_FLAGS);
```
What do you think about using the NFCT_IRQ_FLAGS definition?
This way we avoid conditional compilation at this place.